functions/Get-DbaPageFileSetting.ps1

```diff
@@ -0,0 +1,136 @@
+
+#>[CmdletBinding()]
+	param (
+		[Parameter(Mandatory = $false, Position = 0, ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true)]
+		[Alias("cn", "host", "ServerInstance", "Server", "SqlServer")]
+		[object]$ComputerName,
+		[PSCredential] [System.Management.Automation.CredentialAttribute()]$Credential
+	)
+	BEGIN {}
+	PROCESS {
+        foreach ($Computer in $ComputerName)
+        {
+            Write-Verbose "Connecting to $Computer"
+			$reply = Resolve-DbaNetworkName -ComputerName $Computer -erroraction silentlycontinue
+            if ( $reply.ComputerName )
+            {
+                $computer = $reply.ComputerName
+            }
+            Write-Verbose "Connecting to $Computer via CIM (WSMan)"
+            $CompSys = Get-CimInstance -ComputerName $Computer -Query "SELECT * FROM win32_computersystem" -ErrorVariable $MyErr -Credential $Credential
+            if ( $CompSys )
+            {
+                Write-Verbose "Successfully retrieved PageFile information on $Computer via WSMan"
+                if ($CompSys.automaticmanagedpagefile -eq $False)
+                {
+                    $PF = Get-CimInstance -ComputerName $Computer -Query "SELECT * FROM win32_pagefile" # deprecated !
+                    $PFU = Get-CimInstance -ComputerName $Computer -Query "SELECT * FROM win32_pagefileUsage"
+                    $PFS = Get-CimInstance -ComputerName $Computer -Query "SELECT * FROM win32_pagefileSetting"
+                }
+            }
+			else
+            {
+				Write-Verbose "No WSMan connection to $Computer"
+                Write-Verbose "Getting computer information from server $Computer via CIM (DCOM)"
+			    $sessionoption = New-CimSessionOption -Protocol DCOM
+			    $CIMsession = New-CimSession -ComputerName $Computer -SessionOption $sessionoption -Credential $Credential
+                $CompSys = Get-CimInstance -CimSession $CIMsession -Query "SELECT * FROM win32_computersystem" -ErrorVariable $MyErr
+                if ( $CompSys )
+                {
+                    if ( $CompSys.automaticmanagedpagefile )
+                    {
+                        Write-Verbose "Successfully retrieved PageFile information on $Computer via DCom"
+                        if ($CompSys.automaticmanagedpagefile -eq $False)
+                        {
+                            $PF = Get-CimInstance -CimSession $CIMsession -Query "SELECT * FROM win32_pagefile" # deprecated !
+                            $PFU = Get-CimInstance -CimSession $CIMsession -Query "SELECT * FROM win32_pagefileUsage"
+                            $PFS = Get-CimInstance -CimSession $CIMsession -Query "SELECT * FROM win32_pagefileSetting"
+                        }
+                    }
+                    else
+                    {
+                    Write-Warning "$computer Operating System too old. No Pagefile information available."

```

Comment:
I don't have changed anything in the pagefile settings.
I want to test the ELSE block and I expected that it will work like the IF (1st screenshot).
It is not suppose? I tested on sql2008 machine.